### DISCUSSION TITLE
Use some ID3 tag to identify a track in a database

### DISCUSSION THREAD

**Discussion by Baz93** [navidrome/navidrome#3151] [Ideas]
Currently, changing a track's file path results in its removal from the database. To prevent this, I propose leveraging an ID3 tag, such as "NAVIDROME_ID" or "TRACK_ID," which users can pre-generate uniquely for each track. If this tag is present in the file, it will serve as the identifier; otherwise, the file path will continue to be used.

**Comment by metalheim** [comment]
Something like this is actually planned in the BFR #2709 but not with a custom tag, but rather using tags and/or Musicbrainz IDs (that many people have present in their files already).

When this big refactor is released, the idea is to be persistent against files moves by:

Songs:
- MBID OR 
- AlbumID+DiscNumber+TrackNumber OR 
- AlbumID+TrackTitle

Album:
- MBID OR 
- AlbumName+AlbumVersion+ReleaseDate
Artist: 
- ArtistName (will use MBID as well)


https://github.com/navidrome/navidrome/blob/dcbefb9e3be2625280f7bcef92d33885dfad134c/model/metadata/persistent_ids.go

**Reply by Baz93** [reply]
But what if I change the MBID? Sometimes, I might select the wrong recording of the same song by mistake or because it was not present on MusicBrainz. Later, I may want to update the tag to the correct one. Additionally, it's possible to have two identical tracks in different albums in the library. In such cases, MBID might not be sufficient. Even Acoustic ID might not solve this issue. A custom tag would offer more flexibility and accuracy in identifying tracks, as it allows for a unique, persistent identifier that remains consistent regardless of other metadata changes. In my opinion, a custom tag is a much better option for ensuring reliable track identification and persistence.

**Reply by deluan** [reply]
> A custom tag would offer more flexibility and accuracy in identifying tracks, as it allows for a unique, persistent identifier that remains consistent regardless of other metadata changes. In my opinion, a custom tag is a much better option for ensuring reliable track identification and persistence.

@Baz93 when designing the persistent ID functionality in the new scanner (aka BFR), I thought about a solution exactly like this. But this is not something that most users are willing to do. 

But don't worry, the solution will be something very similar to what @metalheim posted, but the tag(s) used for creating the PID will be configurable, so you will be able to do what you want after a quick configuration.

> Additionally, it's possible to have two identical tracks in different albums in the library. In such cases, MBID might not be sufficient. 

In this case, the songs have identical `musicbrainz_recording_id`, but distinct `musicbrainz_release_track_id`, which is the MBID used by Navidrome, so it wouldn't be an issue.
